src/mrinufft/io/siemens.py

```diff
@@ -0,0 +1,87 @@
+import numpy as np
+
+try:
+    from mapvbvd import mapVBVD
+    MAPVBVD_FOUND = True
+except ImportError:
+    MAPVBVD_FOUND = False
+
+
+def read_rawdat(filename: str, removeOS: bool = False, squeeze: bool = True,
+                data_type: str = "SPARKLING_VE11C"):
+    """Read raw data from a Siemens MRI file.
+    
+    Parameters
+    ----------
+    filename : str
+        The path to the Siemens MRI file.
+    removeOS : bool, optional
+        Whether to remove the oversampling, by default False.
+    squeeze : bool, optional
+        Whether to squeeze the dimensions of the data, by default True.
+    data_type : str, optional
+        The type of data to read, by default 'SPARKLING_VE11C'.
+
+    Returns
+    -------
+    tuple
+        A tuple containing the raw data and the header information.
+
+    Raises
+    ------
+    ImportError
+        If the mapVBVD module is not available.
+
+    Notes
+    -----
+    This function requires the mapVBVD module to be installed. 
+    You can install it using the following command:
+        `pip install pymapVBVD`
+    """
+    if not MAPVBVD_FOUND:
+        raise ImportError(
+            "The mapVBVD module is not available. Please install it using "
+            "the following command: pip install pymapVBVD"
+        )
+    twixObj = mapVBVD(filename)
+    if isinstance(twixObj, list):
+        twixObj = twixObj[-1]
+    twixObj.image.flagRemoveOS = removeOS
+    twixObj.image.squeeze = squeeze
+    raw_kspace = twixObj.image['']
+    data = np.moveaxis(raw_kspace, 0, 2)
+    hdr = {
+        "num_coils": int(twixObj.image.NCha),
+        "num_shots": int(twixObj.image.NLin),
+        "num_contrasts": int(twixObj.image.NSet),
+        "num_adc_samples": int(twixObj.image.NCol),
+        "num_slices": int(twixObj.image.NSli),
+    }
+    data = data.reshape(
+        hdr["num_coils"], 
+        hdr["num_shots"]*hdr["num_adc_samples"], 
+        hdr["num_slices"], 
+        hdr["num_contrasts"]
+    )
+    if "SPARKLING_VE11C" in data_type:
```

Comment:
do you have other value of data_type in mind ? 
I would recommend to use an enum to gather them. Overall this section of the code only updates the header, and may be refactored to a dedicated function. The workflow would then be 

```python 
parser = parse_header_type[data_type] # e.g. using a `MethodRegister` from `_utils.py`
hdr |= parser(twixObj)
```